Mathlib/LinearAlgebra/RootSystem/Defs.lean

```diff
@@ -103,45 +118,195 @@ structure RootSystem extends RootPairing ι R M N :=
 namespace RootPairing
 
 variable {ι R M N}
-variable (P : RootPairing ι R M N) (i : ι)
+variable (P : RootPairing ι R M N) (i j : ι)
 
-/-- A root pairing is said to be crystallographic if the pairing between a root and coroot is
-always an integer. -/
-def IsCrystallographic : Prop :=
-  ∀ i, MapsTo (P.toLin (P.root i)) (range P.coroot) (zmultiples (1 : R))
+lemma eq_of_root (h : P.root i = P.root j) : i = j :=
+  (EmbeddingLike.apply_eq_iff_eq P.root).mp h
 
-/-- A root pairing is said to be reduced if any linearly dependent pair of roots is related by a
-sign. -/
-def IsReduced : Prop :=
-  ∀ i j, ¬ LinearIndependent R ![P.root i, P.root j] → (P.root i = P.root j ∨ P.root i = - P.root j)
+lemma root_ne (h: i ≠ j) : P.root i ≠ P.root j :=
+  fun h' ↦ h (eq_of_root P i j h')
```

Comment:
```suggestion
  P.root.injective.ne h
```
Not sure if it is worth having these lemmas, given that the proofs are so concise they could be inlined.
I'll leave that up to you.

Reply:
Agreed, I haven't forgotten this PR belongs to @ScottCarnahan but I hope he won't mind if I make the call to cut these two.